api/src/guards/api-key.guard.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+
+@Injectable()
+export class ApiKeyGuard implements CanActivate {
+  canActivate(context: ExecutionContext) {
+    const req = context.switchToHttp().getRequest();
+
+    if (
```

Comment:
Should we also be throwing if the pass key is not set?

Reply:
It'd probably be best to keep it optional, especially on doorway there will probably be a window where the env variable hasn't been set yet and we'll want to allow requests through while that is set up/spinning up